mysql-async/src/main/java/com/github/jasync/sql/db/mysql/pool/MySQLConnectionFactory.kt

```diff
@@ -29,7 +26,17 @@ open class MySQLConnectionFactory(val configuration: Configuration) : Connection
      * @return
      */
     override fun create(): CompletableFuture<MySQLConnection> {
-        val connection = MySQLConnection(configuration = configuration)
-        return connection.connect()
+        return configuration.credentialsProvider
```

Comment:
I think instead of this change the connection to use credentialsProvider

Reply:
This has some implications regarding refactoring, as the connection currently expects the configuration to be fully resolved on creation, eg. it passes the configuration to the `ConnectionHandler` which in turn expects the username to be present for the handshake message. Given the reactive nature of the `CredentialsProvider`, functionality like this would have to be moved from the constructor to the `onConnect` method. It would probably also cause the change of some public method signatures which are however not part of `Connection` / `ConcreteConnection` interface, so might be OK to do it.

This seems like significant refactoring to me, so I'd like to make sure it's what we want before committing time to it :)